### PR TITLE
New chains demo scripts including a quay.io push

### PR DESCRIPTION
**Pull request by simonbaird**
Replaces https://github.com/simonbaird/infra-deployments/pull/3 .

### CODE REVIEW

**Comment by simonbaird** [comment]
FYI @caugello it's based on https://docs.engineering.redhat.com/display/~caugello/2022/02/21/HACBS+-+Tekton+Signed+Provenance+-+Quay.io with minor tweaks so we can pass in a secret name.

But I didn't get the verification working. Not sure what I'm missing. If you want to try to fix it please feel free.

Update: Disregard the above. It does work - I just forgot the `kubectl patch sa pipeline ...`

**Comment by simonbaird** [comment]
Probably ready for review now. Rather than review the code it's probably more useful to see if you can run the demos locally.

File: hack/chains/pipeline-quay-demo.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+pause
+
+title "Creating and configuring pipeline for demo"
+kubectl apply -f $SCRIPTDIR/pipeline-quay-demo.yaml
+
+title "Adding secret to tekton-chains service account"
+# Chains also needs the quay secret so it can push signatures
+# and (in future maybe) attestations
+kubectl patch sa pipeline -n tekton-chains \
```
Comment (caugello) [inline hack/chains/pipeline-quay-demo.sh:50]: Not blocking, do we need to have the demo tied up to the tekton-chains project ?

**Comment by caugello** [comment]
LGTM. could run the demo.

File: hack/chains/pipeline-quay-demo.sh (hunk 1 above)
Comment (simonbaird) [reply hack/chains/pipeline-quay-demo.sh:50]: For now yes, but I have a todo somewhere to make it work in any project

**Comment by simonbaird** [comment]
Long story: I updated contract-devel with the latest from main branch, but then I found the gitops sync would not complete fully.

Rather than debug that, I rewound it back to where were before. This way we can have a more stable branch for the demo. Will need to figure out why the sync is failing, but let's ignore it for now.

**Comment by simonbaird** [comment]
As discussed at standup - I think this is stable enough now, let's merge it.

**Pull request merged**